contentheader/README.md

```diff
@@ -0,0 +1,96 @@
+
+This JupyterLab example extension is intended to demo one specific feature of `MainAreaWidget`, namely, its `contentHeader` section.
+
+> As background, `MainAreaWidget` is a high-level JupyterLab widget that conventionally is used to enclose the Launcher (shown above) or a notebook editor. The `contentHeader`, in turn, is a Lumino `BoxPanel` widget positioned at the very top of this main area. This makes the `contentHeader` potentially useful to extensions needing some place to put content that the user will _always see_.
+
+In code: after we get a `MainAreaWidget`, for example via
+
+```ts
+// src/index.ts#L37-L37
+
+const main = app.shell.currentWidget;
+```
+
+we can then create a widget of interest, for example as
+
+```ts
+// src/index.ts#L40-L40
+
+const widget = new Widget();
+```
+
+before finally adding it to the JupyterLab main area's `contentHeader` real estate at its very top:
+
+```ts
+// src/index.ts#L45-L45
+
+main.contentHeader.addWidget(widget);
+```
+
+## Install
+
+First, ensure you've followed the installation instructions in the [top-level README](../README.md), e.g.:
+
+```bash
+# clone the repository
+git clone https://github.com/jupyterlab/extension-examples.git jupyterlab-extension-examples
+
+# go to the extension examples folder
+cd jupyterlab-extension-examples
+
+# create a new environment
+conda env create
+
+# activate the environment
+conda activate jupyterlab-extension-examples
+```
+
+Then build this extension and launch it—this largely also follows from the instructions in the top-level [README](../README.md), except using this repo instead of the `hello-world`:
+
+```bash
+# go to the contentheader example
+cd contentheader
+
+# install the extension in editable mode
+python -m pip install -e .
+
+# install your development version of the extension with JupyterLab
+jupyter labextension develop . --overwrite
+
+# build the TypeScript source after making changes
+jlpm run build
+
+# start JupyterLab
+jupyter lab
+```
+
+## Activate
+
+As in the demo animated GIF above, once in JupyterLab
+
+1. open the [Command Palette](https://jupyterlab.readthedocs.io/en/stable/user/commands.html) via, e.g., View ➜ Activate Command Palette.
+2. Type in "populate" to see the command created by this extension: "Populate content header (time example)".
+3. This will create a small header bar at the top of the main JupyterLab window that shows the current time in GMT (Greenwich Mean Time, also called UTC, Coordinated Universal Time).
+
+> Nota bene, you can _toggle_ the `contentHeader` via the Command Palette ➜ "Show Header Above Content" if you decide you don't want to see it.
+
+> Nota bene 2, the `contentHeader` and the "Show Header Above Content" toggle apply on a _per_ `MainAreaWidget`: if you run this extension, it will show the time in the `contentHeader` of the _active_ main area, and the time will not appear in another notebook tab. This may be useful for some extension workflows and not others.
```

Comment:
```suggestion
> Nota bene 2, the `contentHeader` and the _Show Header Above Content_ toggle apply on a _per_ `MainAreaWidget`: if you run this extension, it will show the time in the `contentHeader` of the _active_ main area, and the time will not appear in another notebook tab. This may be useful for some extension workflows and not others.
```